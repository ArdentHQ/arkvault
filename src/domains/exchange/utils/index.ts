export * from "./assertions";
export * from "./errors";
