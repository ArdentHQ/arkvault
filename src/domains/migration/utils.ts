import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
