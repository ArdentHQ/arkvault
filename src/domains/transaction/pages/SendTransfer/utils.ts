export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
