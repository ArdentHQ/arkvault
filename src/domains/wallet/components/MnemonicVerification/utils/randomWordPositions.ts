import { sample } from "@ardenthq/sdk-helpers";

const randomWordPositions = (length: number): number[] => {
	const positions: number[] = [...Array.from({ length }).keys()];
	const result: number[] = [];

	while (result.length < 3) {
		const randomNumber = sample(positions) + 1;

		if (result.includes(randomNumber)) {
			continue;
		}

		result.push(randomNumber);
	}

	return result.sort((a, b) => a - b);
};

export { randomWordPositions };
