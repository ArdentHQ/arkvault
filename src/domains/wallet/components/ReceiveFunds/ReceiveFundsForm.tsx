import { Networks } from "@ardenthq/sdk";
import React, { useEffect } from "react";
import { useFormContext } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { FormField, FormHelperText, FormLabel } from "@/app/components/Form";
import { InputCurrency } from "@/app/components/Input";

export const ReceiveFundsForm = () => {
	const { t } = useTranslation();

	const form = useFormContext();
	const { getValues, setValue, register } = form;

	useEffect(() => {
		register("amount");
	}, [register]);

	return (
		<div data-testid="ReceiveFundsForm">
			<div className="mt-4">
				<FormField name="amount">
					<FormLabel label={t("COMMON.AMOUNT")} optional />
					<InputCurrency
						data-testid="ReceiveFundsForm__amount"
						placeholder={t("COMMON.AMOUNT")}
						className="pr-20"
						value={getValues("amount")}
						onChange={(amount) => setValue("amount", amount)}
					/>
					<FormHelperText />
				</FormField>
			</div>
		</div>
	);
};
