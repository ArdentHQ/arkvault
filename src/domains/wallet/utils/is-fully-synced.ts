import { Contracts } from "@payvo/sdk-profiles";

export const isFullySynced = (wallet: Contracts.IReadWriteWallet): boolean =>
	wallet.hasBeenFullyRestored() && (wallet.hasSyncedWithNetwork() || wallet.balance() === 0);
