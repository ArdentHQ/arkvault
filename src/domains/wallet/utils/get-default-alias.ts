import { Networks } from "@ardenthq/sdk";
import { Contracts } from "@ardenthq/sdk-profiles";
import { networkDisplayName } from "@/utils/network-utils";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${networkDisplayName(network)} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
