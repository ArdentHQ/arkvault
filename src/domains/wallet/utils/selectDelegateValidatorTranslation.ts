import { Networks } from "@ardenthq/sdk";
import { isMainsailNetwork } from "@/utils/network-utils";

interface Properties<T extends string | React.ReactNode> {
	network: Networks.Network;
	validatorStr: T;
	delegateStr: T;
}

export const selectDelegateValidatorTranslation = <T extends string | React.ReactNode>({ network, validatorStr, delegateStr }: Properties<T>): T => isMainsailNetwork(network) ? validatorStr : delegateStr;



