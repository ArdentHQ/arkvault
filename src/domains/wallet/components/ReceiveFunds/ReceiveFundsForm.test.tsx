/* eslint-disable @typescript-eslint/require-await */
import { Networks } from "@payvo/sdk";
import userEvent from "@testing-library/user-event";
import React from "react";

import { ReceiveFundsForm } from "./ReceiveFundsForm";
import { env, getDefaultProfileId, getDefaultWalletId, renderWithForm, screen, waitFor } from "@/utils/testing-library";

let network: Networks.Network;

describe("ReceiveFundsForm", () => {
	beforeEach(() => {
		const profile = env.profiles().findById(getDefaultProfileId());
		const wallet = profile.wallets().findById(getDefaultWalletId());
		network = wallet.network();
	});

	it("should render", async () => {
		const { asFragment } = renderWithForm(<ReceiveFundsForm network={network} />);

		await waitFor(() => expect(screen.getByTestId("ReceiveFundsForm__amount")).not.toHaveValue());
		await waitFor(() => expect(screen.getByTestId("ReceiveFundsForm__memo")).not.toHaveValue());

		expect(asFragment()).toMatchSnapshot();
	});

	it("should emit amount onChange event", async () => {
		const { asFragment, form } = renderWithForm(<ReceiveFundsForm network={network} />);

		await waitFor(() => expect(screen.getByTestId("ReceiveFundsForm__amount")).not.toHaveValue());

		userEvent.paste(screen.getByTestId("ReceiveFundsForm__amount"), "10");

		await waitFor(() => expect(form()?.getValues("amount")).toBe("10"));

		expect(asFragment()).toMatchSnapshot();
	});

	it("should emit memo onChange event", async () => {
		const { asFragment, form } = renderWithForm(<ReceiveFundsForm network={network} />);
		await waitFor(() => expect(screen.getByTestId("ReceiveFundsForm__memo")).not.toHaveValue());

		userEvent.paste(screen.getByTestId("ReceiveFundsForm__memo"), "test");
		await waitFor(() => expect(form()?.getValues("memo")).toBe("test"));
		await waitFor(() => expect(screen.getByTestId("ReceiveFundsForm__memo")).toHaveValue("test"));

		expect(asFragment()).toMatchSnapshot();
	});

	it("should not show memo if is not supported by network", async () => {
		const memo = Array.from({ length: 256 }).fill("x").join("");

		const memoMock = jest.spyOn(network, "usesMemo").mockReturnValue(false);

		const { asFragment } = renderWithForm(<ReceiveFundsForm network={network} />, {
			defaultValues: { memo },
		});

		expect(screen.queryByTestId("ReceiveFundsForm__memo")).not.toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();

		memoMock.mockRestore();
	});
});
