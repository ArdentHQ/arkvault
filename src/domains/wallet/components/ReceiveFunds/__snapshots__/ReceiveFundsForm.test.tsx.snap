// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReceiveFundsForm > should emit amount onChange event 1`] = `
<DocumentFragment>
  <div
    data-testid="ReceiveFundsForm"
  >
    <div
      class="mt-4"
    >
      <fieldset
        class="[&:focus-within_.FormLabel]:text-theme-secondary-text dark:[&:focus-within_.FormLabel]:text-theme-primary-500 [&>.FormLabel]:text-theme-secondary-text dark:[&>.FormLabel]:text-theme-secondary-500 flex min-w-0 flex-col"
      >
        <label
          class="FormLabel mb-2 flex text-sm font-semibold leading-[17px] text-theme-secondary-text transition-colors duration-100"
          data-testid="FormLabel"
          for="amount"
        >
          Amount
          <span
            class="ml-1 text-theme-secondary-500 dark:text-theme-secondary-700"
            data-testid="FormLabel__optional"
          >
            optional
          </span>
        </label>
        <div
          class="relative"
        >
          <div
            class="flex w-full appearance-none items-center space-x-2 rounded px-4 text-theme-text transition-colors duration-200 border border-theme-secondary-400 bg-theme-background focus-within:border-theme-primary-600 focus-within:ring-theme-primary-600 dark:border-theme-secondary-700 focus-within:ring-1 h-12 overflow-hidden sm:h-14 pr-20"
          >
            <div
              class="relative flex h-full flex-1"
            >
              <input
                autocomplete="off"
                class="!bg-transparent !p-0 focus:shadow-none focus:outline-none focus:!ring-0 focus:!ring-transparent [&.shadow-none]:shadow-none no-ligatures w-full border-none !text-sm placeholder:text-theme-secondary-400 dark:placeholder:text-theme-secondary-700 sm:!text-base"
                data-testid="ReceiveFundsForm__amount"
                name="amount"
                placeholder="Amount"
                type="text"
                value="10"
              />
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReceiveFundsForm > should render 1`] = `
<DocumentFragment>
  <div
    data-testid="ReceiveFundsForm"
  >
    <div
      class="mt-4"
    >
      <fieldset
        class="[&:focus-within_.FormLabel]:text-theme-secondary-text dark:[&:focus-within_.FormLabel]:text-theme-primary-500 [&>.FormLabel]:text-theme-secondary-text dark:[&>.FormLabel]:text-theme-secondary-500 flex min-w-0 flex-col"
      >
        <label
          class="FormLabel mb-2 flex text-sm font-semibold leading-[17px] text-theme-secondary-text transition-colors duration-100"
          data-testid="FormLabel"
          for="amount"
        >
          Amount
          <span
            class="ml-1 text-theme-secondary-500 dark:text-theme-secondary-700"
            data-testid="FormLabel__optional"
          >
            optional
          </span>
        </label>
        <div
          class="relative"
        >
          <div
            class="flex w-full appearance-none items-center space-x-2 rounded px-4 text-theme-text transition-colors duration-200 border border-theme-secondary-400 bg-theme-background focus-within:border-theme-primary-600 focus-within:ring-theme-primary-600 dark:border-theme-secondary-700 focus-within:ring-1 h-12 overflow-hidden sm:h-14 pr-20"
          >
            <div
              class="relative flex h-full flex-1"
            >
              <input
                autocomplete="off"
                class="!bg-transparent !p-0 focus:shadow-none focus:outline-none focus:!ring-0 focus:!ring-transparent [&.shadow-none]:shadow-none no-ligatures w-full border-none !text-sm placeholder:text-theme-secondary-400 dark:placeholder:text-theme-secondary-700 sm:!text-base"
                data-testid="ReceiveFundsForm__amount"
                name="amount"
                placeholder="Amount"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
